### PR TITLE
[SPARK-47743][Core] Use milliseconds as the time unit in logging

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Use milliseconds as the time unit in logging

### Why are the changes needed?

During migrations, for the same log key `TIMEOUT` there are [log entry using seconds](https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala#L145) and [another entry using ms](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/executor/Executor.scala#L1017).
I suggest unify them and use milliseconds for all time related loggings.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Exising UT

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @panbingkun

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala [hunk 1]
```diff
@@ -102,7 +103,7 @@ object LogKey extends Enumeration {
   val THREAD_NAME = Value
   val TID = Value
   val TIMEOUT = Value
-  val TIME_UNITS = Value
+  val TOTAL_TIME = Value
```
Comment (gengliangwang) [inline common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala:106]: cc @dtenedor I renamed TIME_UNITS as `TOTAL_TIME` and `EFFECTIVE_TIME`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala [hunk 2]
```diff
@@ -88,15 +88,15 @@ class PlanChangeLogger[TreeType <: TreeNode[_]] extends Logging {
...
-    val totalTime = metrics.time / NANOS_PER_SECOND.toDouble
-    val totalTimeEffective = metrics.timeEffective / NANOS_PER_SECOND.toDouble
+    val totalTime = metrics.time / NANOS_PER_MILLIS.toDouble
+    val totalTimeEffective = metrics.timeEffective / NANOS_PER_MILLIS.toDouble
     val message: MessageWithContext =
       log"""
          |=== Metrics of Executed Rules ===
          |Total number of runs: ${MDC(RULE_NUMBER_OF_RUNS, metrics.numRuns)}
-         |Total time: ${MDC(TIME_UNITS, totalTime)} seconds
+         |Total time: ${MDC(EFFECTIVE_TIME, totalTime)} ms
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala:97]: cc @dtenedor

**Comment by gengliangwang** [comment]
Merging to master to unblock other migration PRs.

**Pull request closed without merging**